lib/core/ast_processors/okl_sema_processor/processor.cpp

```diff
@@ -7,11 +7,44 @@
 #include "core/ast_processors/okl_sema_processor/handlers/loop.h"
 
 #include <clang/AST/AST.h>
+#include <spdlog/spdlog.h>
 
 namespace {
 using namespace clang;
 using namespace oklt;
 
+namespace {
+HandleResult semaDefaultPre(const clang::Attr*,
+                            const clang::Stmt& stmt,
+                            OklSemaCtx& sema,
+                            SessionStage& stage) {
+    switch (stmt.getStmtClass()) {
+        case clang::Stmt::ForStmtClass: {
+            auto& forStmt = *dyn_cast<ForStmt>(&stmt);
+            return preValidateOklForLoopWithoutAttribute(nullptr, forStmt, sema, stage);
+        }
+        default: {
+        }
+    }
+    return {};
+}
+
+HandleResult semaDefaultPost(const clang::Attr*,
+                             const clang::Stmt& stmt,
+                             OklSemaCtx& sema,
+                             SessionStage& stage) {
+    switch (stmt.getStmtClass()) {
```

Comment:
This one is better be simplified to `if (auto forStmt = dyn_cast_or_null<ForStmt>(&stmt))`

Reply:
Good idea